linkerd/app/outbound/src/ingress.rs

```diff
@@ -267,8 +242,126 @@ impl Outbound<svc::ArcNewHttp<http::Endpoint>> {
     }
 }
 
-#[derive(Clone, Debug)]
-struct SelectTarget(Http<tcp::Accept>);
+// === impl Http ===
+
+impl<T> Param<http::Version> for Http<T> {
+    fn param(&self) -> http::Version {
+        self.version
+    }
+}
+
+impl Param<http::normalize_uri::DefaultAuthority> for Http<tcp::Accept> {
+    fn param(&self) -> http::normalize_uri::DefaultAuthority {
+        http::normalize_uri::DefaultAuthority(Some(
+            self.target
+                .orig_dst
+                .to_string()
+                .parse()
+                .expect("Address must be a valid authority"),
+        ))
+    }
+}
+
+impl Param<profiles::LookupAddr> for Http<NameAddr> {
+    fn param(&self) -> profiles::LookupAddr {
+        profiles::LookupAddr(self.target.clone().into())
+    }
+}
+
+impl Param<Remote<ServerAddr>> for Http<OrigDstAddr> {
+    fn param(&self) -> Remote<ServerAddr> {
+        let OrigDstAddr(addr) = self.target;
+        Remote(ServerAddr(addr))
+    }
+}
+
+impl Param<tls::ConditionalClientTls> for Http<OrigDstAddr> {
+    fn param(&self) -> tls::ConditionalClientTls {
+        tls::ConditionalClientTls::None(tls::NoClientTls::NotProvidedByServiceDiscovery)
+    }
+}
+
+impl Param<Option<tcp::opaque_transport::PortOverride>> for Http<OrigDstAddr> {
+    fn param(&self) -> Option<tcp::opaque_transport::PortOverride> {
+        None
+    }
+}
+
+impl Param<Option<http::AuthorityOverride>> for Http<OrigDstAddr> {
+    fn param(&self) -> Option<http::AuthorityOverride> {
+        None
+    }
+}
+
+impl svc::Param<transport::labels::Key> for Http<OrigDstAddr> {
+    fn param(&self) -> transport::labels::Key {
+        transport::labels::Key::OutboundClient(self.param())
+    }
+}
+
+impl svc::Param<metrics::OutboundEndpointLabels> for Http<OrigDstAddr> {
+    fn param(&self) -> metrics::OutboundEndpointLabels {
+        let authority = self
+            .target
+            .to_string()
+            .parse()
+            .expect("address must be a valid authority");
+        metrics::OutboundEndpointLabels {
+            authority: Some(authority),
+            labels: Default::default(),
+            server_id: self.param(),
+            target_addr: self.target.into(),
+        }
+    }
+}
+
+impl svc::Param<metrics::EndpointLabels> for Http<OrigDstAddr> {
+    fn param(&self) -> metrics::EndpointLabels {
+        metrics::EndpointLabels::Outbound(self.param())
+    }
+}
+
+impl svc::Param<http::client::Settings> for Http<OrigDstAddr> {
+    fn param(&self) -> http::client::Settings {
+        match self.param() {
+            http::Version::H2 => http::client::Settings::H2,
+            http::Version::Http1 => http::client::Settings::Http1,
+        }
+    }
+}
+
+// TODO(ver) move this into the endpoint stack?
```

Comment:
is this comment still relevant here? looks like it was pasted from the earlier code?